### ISSUE TITLE
Support for GraphQLInterface type

### ISSUE THREAD

**Issue by lromanowicz** [mirumee/ariadne-codegen#77]
Turned out that some of our types implement interfaces like example below:

schema.graphql

```
interface RecipientPaymentAccount {
  accountNumber: String
  alias: String
  <other_fields>
}
```

```
type DomesticPaymentAccount implements RecipientPaymentAccount {
  accountNumber: String
  alias: String
  <other_fields>
}
```

queries.graphql
```
  query PaymentAccount($paymentAccountId: String!) {
    paymentAccount(paymentAccountId: $paymentAccountId) {
      id
      ... on DomesticPaymentAccount {
         <some_fields_here>
      }
    }
  }
```

In the end, it generates an invalid empty class like below and black throws an error when formatting the code.
Not sure if this example will generate the issue... but that's the only connection I found around it... the `interface` type.

```
class GenericTransactionGenericTransactionRecipientPaymentAccount(BaseModel):
GenericTransaction.update_forward_refs()
GenericTransactionGenericTransaction.update_forward_refs()
```

**Comment by mat-sop** [comment]
I investigated this issue and found out that a problem occurs for a combination of using interface as a field type and inline fragments. In this combination, inline fragments are not handled properly, and fields selected that way are skipped from generated class. If some query selects fields only using inline fragments then generated class is “empty” and black throws an exception.

For example, let`s take a simple schema:
```gql
schema {
  query: Query
}

type Query {
  testQuery: TestInterface
}

interface TestInterface {
  id: String
}

type TestTypeA implements TestInterface {
  id: String
  number: Float
}

type TestTypeB implements TestInterface {
  id: String!
  name: String
}
```
and query which selects all fields:
```gql
query Test {
  testQuery {
    id
    ... on TestTypeA {
      number
    }
    ... on TestTypeB {
      name
    }
  }
}
```
Generated class has only `id` field, `number` and `name` fields are omitted.
For query without `id` field, generated class will be "empty" and black will throw exception.

Until issue is fixed I found a workaround, but it requires changes in schema:
- Add `union` with all types which implement interface, eg `union WorkaroundTestInterface = TestTypeA | TestTypeB`
- Replace occurrences of interface as field types with created union, eg. `testQuery: TestInterface` -> `testQuery: WorkaroundTestInterface`
